### PR TITLE
feat: Add support of confluence server.

### PR DESCRIPTION
**Pull request by vpommier**
Add support of [personal access token authentication](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html) for confluence server.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -72,6 +73,7 @@ Options:
   -c --config <path>   Use the specified configuration file.
                         [default: $HOME/.config/mark]
   --ci                 Runs on CI mode. It won't fail if files are not found.
+	--confluence-cloud   Use Confluence Cloud or Confluence hosted server authentication method.
```
Comment (kovetskiy) [inline main.go:76]: Something wrong with indentation here

**Review by kovetskiy — CHANGES_REQUESTED** [review]
I'm not sure it's a good way to break everyone's workflow and make them use some new flag which was not required before.

Can we just always send the Authorization token for both cloud and server?

**Comment by vpommier** [comment]
> I'm not sure it's a good way to break everyone's workflow and make them use some new flag which was not required before.
> 
> Can we just always send the Authorization token for both cloud and server?

When i send the Authorization token it works only for server but not cloud.
And even if it worked there is no guarantee that this behavior (having basic auth + bearer token at the same time) will be supported in the future. It is the best way to attract issues in my opinion.

The thing that i can do is to require the new flag only for confluence server, in that way the new version will doesn't break the actual workflow.

I made these kind of changes to rapidly test if the tool can solve my issue with this authentication method. But i think the proper way to fix the code is to patch the code this way:
* [Headers should be copied instead of passing the reference of them `r.Headers.Clone()`. Because it create side effet.](https://github.com/kovetskiy/gopencils/blob/master/resource.go#L54)
* [Provide something more than just basic auth.](https://github.com/kovetskiy/gopencils/blob/master/api.go#L50)

File: main.go (hunk 1 above)
Comment (vpommier) [reply main.go]: Fixed

**Comment by vpommier** [comment]
Hi @kovetskiy i pushed modifications any news please ?

**Comment by kovetskiy** [comment]
Works great, thanks!

**Pull request merged**